bindings/ruby/lib/fdbtuple.rb

```diff
@@ -182,15 +198,34 @@ def self.encode(v, nested=false)
           raise ArgumentError, "unsupported encoding #{v.encoding.name}"
         end
       elsif v.kind_of? Integer
-        raise RangeError, "value outside inclusive range -2**64+1 to 2**64-1" if v < -2**64+1 || v > 2**64-1
+        raise RangeError, "Integer magnitude is too large (more than 255 bytes)" if v < -2**2040+1 || v > 2**2040-1
         if v == 0
           @@INT_ZERO_CODE.chr
         elsif v > 0
-          n = bisect_left( @@size_limits, v )
-          (20+n).chr + [v].pack("Q>").slice(8-n, n)
+          if v > @@size_limits[-1]
+            length = (v.bit_length + 7) / 8
+            result = @@POS_INT_END.chr + length.chr
+            length.times do |i|
+              result << ((v >> (8 * (length-i-1))) & 0xff)
+            end
+            result
+          else
+            n = bisect_left( @@size_limits, v )
+            (@@INT_ZERO_CODE+n).chr + [v].pack("Q>").slice(8-n, n)
+          end
         else
-          n = bisect_left( @@size_limits, -v )
-          (20-n).chr + [@@size_limits[n]+v].pack("Q>").slice(8-n, n)
+          if -v > @@size_limits[-1]
+            length = ((-v).bit_length + 7) / 8
+            v += (1 << (length * 8)) - 1
+            result = @@NEG_INT_START.chr + (length ^ 0xff).chr
+            length.times do |i|
+              result << ((v >> (8 * (length-i-1))) & 0xff)
```

Comment:
I have a similar comment about how it's weird that we have to do this ourselves. It also took me a while to figure out that the `<<` operator here means "write this byte to this byte array" and the `>>` means "right shift this integer". I don't think there's much that can be done about that, though.

Reply:
Similarly, I couldn't find a better way of doing this myself.